src/f32-rsum/gen/f32-rsum-rvv-u4v.c

```diff
@@ -0,0 +1,52 @@
+// Auto-generated file. Do not edit!
+//   Template: src/f32-rsum/rvv.c.in
+//   Generator: tools/xngen
+//
+// Copyright 2024 Imagination Technologies, Inc.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include <assert.h>
+
+#include <xnnpack/common.h>
+#include <xnnpack/reduce.h>
+#include <riscv_vector.h>
+
+void xnn_f32_rsum_ukernel__rvv_u4v(
+    size_t batch,
+    const float* input,
+    float* output,
+    const union xnn_f32_scale_params params[restrict XNN_MIN_ELEMENTS(1)])
+{
+  assert(batch != 0);
+  assert(batch % sizeof(float) == 0);
+  assert(input != NULL);
+  assert(output != NULL);
+
+  batch >>= XNN_LOG2_SIZEOF_FLOAT;
+  vfloat32m1_t acc_f32v = __riscv_vfmv_s_f_f32m1(0.f, 1);
+  size_t n = __riscv_vsetvl_e32m4(batch);
+  vfloat32m4_t sum0_f32v = __riscv_vfmv_s_f_f32m4(0.f, n);
+  vfloat32m4_t sum1_f32v = __riscv_vfmv_s_f_f32m4(0.f, n);
+  for (; batch >= n * 4; batch -= n * 4) {
+    vfloat32m4_t in0_f32v = __riscv_vle32_v_f32m4(input, n); input += n;
+    vfloat32m4_t in1_f32v = __riscv_vle32_v_f32m4(input, n); input += n;
+    vfloat32m4_t in2_f32v = __riscv_vle32_v_f32m4(input, n); input += n;
+    vfloat32m4_t in3_f32v = __riscv_vle32_v_f32m4(input, n); input += n;
+    vfloat32m4_t sum01_f32v = __riscv_vfadd_vv_f32m4(in0_f32v, in1_f32v, n);
```

Comment:
Would it be more efficient to use __riscv_vfredosum_vs_f32m4_f32m1 in the loop?

Reply:
No, In unrolled loop we are using 2 accumulators for better stall management and afterwards using __riscv_vfredosum_vs_f32m4_f32m1.